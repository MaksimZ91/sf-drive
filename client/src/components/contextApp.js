import React, { useState }  from 'react'
import { useAuth } from '../hooks/autn.hook'


export const FormContex = React.createContext()


export const FormProvider = ({ children }) => {
  const [openAuthor, setOpenAuthor]=useState(false)
  const [recovery, setRecovery]=useState(false)
  const {login,accessToken, refreshToken, userId, setAccessToken}=useAuth()
  const isAuthen = !!accessToken
   
return (
  <FormContex.Provider value={{openAuthor,setOpenAuthor,recovery, setRecovery,login,refreshToken,userId,isAuthen, setAccessToken, accessToken }}>
    {children}
  </FormContex.Provider>
  )
}
