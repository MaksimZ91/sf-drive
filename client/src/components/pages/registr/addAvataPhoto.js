import React, { useEffect, useRef, useState } from 'react'
import axios, { CancelToken } from 'axios'
import { useHttp } from '../../../hooks/http.hook'
import { CircularProgressbarWithChildren, buildStyles } from 'react-circular-progressbar';
import { useDispatch, useSelector } from 'react-redux'
import { showAlert } from '../../../../redux/actions/actions';
import { deleteAvatarPhoto } from '../../../../redux/actions/userAction';



function AddAvatarPhoto (props){
    const { request } = useHttp()   
    const dispatch = useDispatch() 
    const [photoUrl, setPhotoUrl] = useState(null)
    const [procentUpload, setProcentUpload] = useState(0)   
    const [error, setError] = useState(false)
    const [ loading, setLoading ] = useState(false)  
    const cancelFileUpload = useRef(null)   
    const userAvatar = useSelector((state)=>{
        return { avatar:state.user.userAvatar, avatarName:state.user.avatarPhotoName }
    })
    const urlUpload ='http://localhost:5000/user/upload'
    const urlDelete = 'http://localhost:5000/user/delete-image/'


   

    const onDropHandler = e =>{        
        e.preventDefault()       
        let files = [...e.dataTransfer.files]     
        files.forEach(foto => dispatch(props.value(foto)))
     }

     const handleFileInput = e =>{
        e.preventDefault()
        let files =[...e.target.files]
        files.forEach(foto => dispatch(props.value(foto)))
    }

     const onDragStartHandler = e =>{
         e.preventDefault()
     }
     
     const onDragLeveHandler = e =>{
         e.preventDefault()
     }


    const uploadPhoto = async () => { 
        setLoading(true)
        setError(false)    
        const formData = new FormData()
        formData.append( 'file', userAvatar.avatar)   
        const config ={       
            method:'POST', 
            onUploadProgress: progressEvent => {
                const totalLength = progressEvent.lengthComputable ? progressEvent.total : progressEvent.target.getResponseHeader('content-length') || progressEvent.target.getResponseHeader('x-decompressed-content-length');
                if (totalLength) {
                    let progress = Math.round((progressEvent.loaded * 100) / totalLength)                   
                    setProcentUpload(progress)
                }
            },
            cancelToken: new CancelToken( cancel => cancelFileUpload.current = cancel)
        } 

        await axios.post(urlUpload, formData, config)      
       .then(response => {           
        dispatch(props.avatarName(response.data))
        })
        .catch(function (e) {
            console.log(e)
            setError(true)
            dispatch(showAlert('Не удалось загрузить фото'))
        })
        setLoading(false)
    } 


     useEffect(()=>{
         if(userAvatar.avatar){
         let reader = new FileReader()
         reader.readAsDataURL(userAvatar.avatar)
         reader.onloadend = function () {
             setPhotoUrl(reader.result);
           }
           uploadPhoto()
        }
     },[userAvatar.avatar])

     const cancelUpload = () => {
        if (cancelFileUpload.current)
        dispatch(showAlert('Не удалось загрузить фото'))
        setError(true)       
        cancelFileUpload.current('User has canceled the file upload')
    }

    const onDeletePhoto = async () =>{
        dispatch(deleteAvatarPhoto(null))     
        await request(urlDelete + userAvatar.avatar.name,'DELETE')  
        setPhotoUrl(null) 
    }    



    return(
        <>
        <div className='add_avatar_avatarbox'
            onDrop={onDropHandler}
            onDragStart={onDragStartHandler}
            onDragLeave={onDragLeveHandler}
            onDragOver={onDragStartHandler}>
             <label>
             <img className={!photoUrl?'add_avatar_avatarbox_img':'add_avatar_avatarbox_img active'}src={photoUrl?photoUrl:'../src/img/cameraPhoto.svg'} alt='camera_image'/>
             <input type='file' hidden={true} onChange={handleFileInput}/>
             </label>           
                {(procentUpload<=100 && userAvatar.avatar)?<div className='add_avatar_avatarbox_uploader'></div>:''}
                {(procentUpload<=100 && userAvatar.avatar)?<div className='add_avatar_avatarbox_uploader_wrapper'>                                        
                        <CircularProgressbarWithChildren
                        value={procentUpload}
                        strokeWidth={5}
                        styles={buildStyles({                   
                            pathColor: "#FFFFFF",
                            trailColor: "transparent"
                            })}>  
                        {(!error&&loading)?<img src='../src/img/cancel_fetch.svg' onClick={()=>cancelUpload()}/>:''} 
                        {error?<img src='../src/img/upload_arrow.svg' onClick={uploadPhoto}/>:''}
                        {userAvatar.avatarName?<img src='../src/img/delete_avatar.svg' onClick={onDeletePhoto}/>:''}              
                        </CircularProgressbarWithChildren>                   
                </div>:''}
        </div>
        </>
    )
}

export default AddAvatarPhoto