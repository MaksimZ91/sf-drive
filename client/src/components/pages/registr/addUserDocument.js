import React from 'react'
import Photo from '../../photo/Photo'
import Newphoto from '../../photo/newPhoto'
import { useSelector} from 'react-redux'


function AddUserDocument (props){    
   
    const addAutoPhoto = useSelector((state)=>{
        return state.user[props.value]
    })
    
    return(
        <>
        <div className='add_photo_newphoto'>
            {addAutoPhoto.map((e, index) => <Photo value={e} index={index} photoName={props.photoName} url={props.url} deletePhoto={props.deletePhoto} key={e.lastModified+index}/>)}
           <Newphoto addPhoto={props.addPhoto}/>            
        </div>

        </>
    )
}

export default AddUserDocument