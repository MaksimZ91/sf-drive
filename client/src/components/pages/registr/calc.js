let months = {
  January: 0,
  February: 1,
  March: 2,
  April: 3,
  May: 4,
  June: 5,
  July: 6,
  August: 7,
  September: 8,
  October: 9,
  Novermber: 10,
  December: 11
}

let dayInMons =[31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
 
let week = [6, 0, 1, 2, 3, 4, 5];

export function areEqual(date, currentDate) {
  if (!date || !currentDate) return false;

  return (
    date.getFullYear() === currentDate.getFullYear() &&
    date.getMonth() === currentDate.getMonth() &&
    date.getDate() === currentDate.getDate()
  );
}



export  function getDayOfWeek(date) {
 const dayOfWeek = date.getDay();
  return week[dayOfWeek]
}

export function leapYear (year){
  return !((year % 4) || (!(year % 100) && (year % 400)));
}

export function getDayInMonth (date){
  const month = date.getMonth();
  const years = date.getFullYear();
  const daysInMonth = dayInMons[month];
  
  if (leapYear(years) && month === months.February) {
      return daysInMonth + 1;
  } else {   
      return daysInMonth;      
  }
}

 export function monthDay(year, month){
  let date = new Date (year, month)
  let result=[]
  let day=1
   let daysInMonth = getDayInMonth(date)
   let startDay=getDayOfWeek(date)  
   for (let i = 0; i < (daysInMonth+startDay)/7; i++) {
    result[i] = [];
        for (let j = 0; j < 7; j++) {
        if ((i === 0 && j < startDay) || day > daysInMonth) {
            result[i][j] = undefined;
        } else {
            result[i][j] = new Date(year, month, day++);
        }
      }
    }
    
   return result
  }










