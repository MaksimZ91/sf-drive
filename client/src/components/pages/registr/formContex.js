import React, { useState, useEffect }  from 'react'
import {validation} from '../js/validationForm'


export const FormContex = React.createContext()


export const FormProvider = ({ children }) => {
  const [form, setForm]=useState({ fio:'', email:'', password:'', phone:'', date:'', number:'', passDate:'', about:'', cod:'',numberLicense:'', dateLicense:'' })
  const [formValidation, setFormValidation]=useState(true)

    const onChangeInput = event => {
      setForm({ ...form, [event.target.name]: event.target.value })   
  }   

  useEffect(()=>{setFormValidation((validation(form)))},[form])


 const fixDate =(date) => {
    let day =date.getDate()
    let month = date.getMonth()+1
    if (day<10){ 
      day="0"+ day
    }
    if (month<10){
      month = "0"+ month
    }
    const  newDate = day+"."+month+"."+date.getFullYear()
    return setForm({ ...form,  [event.target.name]: newDate})
  }
  


return (
  <FormContex.Provider value={{form, onChangeInput,fixDate,formValidation, setForm }}>
    {children}
  </FormContex.Provider>
  )
}
