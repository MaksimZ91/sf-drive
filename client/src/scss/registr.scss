@import "mixins";
@import "header";
@import "footer";
@import "fonts";
@import "calendarb";

.registr_main{
  display: flex;
  justify-content: center;
  &_servermessage{
    height: 58px;
    width: 100vw;
    background: #EB5757;
    font-family: Roboto;
    font-style: normal;
    font-weight: 500;
    font-size: 16px;
    line-height: 19px;
    text-align: center;
    color: #FFFFFF;
    z-index: 1;
    position: absolute;

  }
}

input:hover{
  border: 1px solid #61A199;
  box-sizing: border-box;
  border-radius: 6px;
} 

input:focus {
  border: 1px solid #61A199;
  outline-color:#61A199;   
}
  .registr_about{
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin-top: 150px;
  &_titel{ 
    align-self:center;
    >p:first-child{
    font-family: Roboto;
    font-style: normal;
    font-weight: normal;
    font-size: 20px;
    line-height: 147.3%;
    text-align: center;
    color: #828282;
    margin:0;
  }
  >p{
    font-family: Roboto;
    font-style: normal;
    font-weight: bold;
    font-size: 64px;
    line-height: 120%;
    text-align: center;
    color: #1B1B1B;
    margin-top: 14px;
    margin-bottom: 52px;
  }
    @include mobile{
    >p:first-child{
      font-size: 12px;
    }
    >p{
      font-weight: 500;
      font-size: 32px;
    }
  }
}  
  &_form{
    width: 679px;
    height: 319px;    
    >p:first-child{
      font-family: Roboto;
      font-style: normal;
      font-weight: normal;
      font-size: 36px;
      line-height: 147.3%;
      color: #1B1B1B;
      margin-left: 0px;
      margin-bottom: 30px;      
    }
    
    >P{
      font-family: Roboto;
      font-style: normal;
      font-weight: normal;
      font-size: 14px;
      line-height: 16px;
      color: #1B1B1B;
      margin-top: 0;
      margin-bottom: 20px;
    }    
    &_text{
      width: 416px;
      height: 44px;
      background: #FFFFFF;
      border: 1px solid #E0E0E0;
      box-sizing: border-box;
      border-radius: 6px;
      margin-left: 229px;
      padding-left: 12px;
    }
    
    &_date{
      width: 155px;
      height: 44px;
      background: #FFFFFF;
      border: 1px solid #E0E0E0;
      box-sizing: border-box;
      border-radius: 6px;
      margin-left: 158px;
      padding-left: 12px;
      background:url(../img/calendar.svg) no-repeat center right 15px;

    }
    &_email{
      width: 416px;
      height: 44px;
      background: #FFFFFF;
      border: 1px solid #E0E0E0;
      box-sizing: border-box;
      border-radius: 6px;
      margin-left: 132px;
      padding-left: 12px;
      &.active{
        border: 2px solid #EB5757;
        outline-color: #EB5757;       
      }      
    }
    

    &_warn{
      display: block;
      font-family: Roboto;
      font-style: normal;
      font-weight: normal;
      font-size: 14px;
      line-height: 16px;
      color: #EB5757;
      margin-left: 260px;
      margin-bottom: 20px;
      &.active{
        display: block;
      }
    }    
    &_phone{      
      width: 155px;
      height: 44px;
      background: #FFFFFF;
      border: 1px solid #E0E0E0;
      box-sizing: border-box;
      border-radius: 6px;
      margin-left: 203px;
      padding-left: 12px;
    }
   
   
    @include mobile{
      width: 335px;
      >p:first-child{
        font-family: Roboto;
        font-style: normal;
        font-weight: 500;
        font-size: 24px;
        line-height: 120%;
        color: #1B1B1B;
      }
      >p{
        font-family: Roboto;
        font-style: normal;
        font-weight: normal;
        font-size: 14px;
        line-height: 16px;
        color: #1B1B1B;
      }
      &_text {
        width: 335px;
        height: 44px;
        background: #FFFFFF;
        border: 1px solid #E0E0E0;
        box-sizing: border-box;
        border-radius: 6px;        
        margin-left: 0px;
      }
      &_date{
        width: 335px;
        height: 44px;
        background: #FFFFFF;
        border: 1px solid #E0E0E0;
        box-sizing: border-box;
        border-radius: 6px;
        margin-left: 0px;
        background:url(../img/calendar.svg) no-repeat center right 15px;
      }
      &_email{
        width: 335px;
        height: 44px;
        background: #FFFFFF;
        border: 1px solid #E0E0E0;
        box-sizing: border-box;
        border-radius: 6px;
        margin-left: 0px;
      }
      &_phone{      
        width: 335px;
        height: 44px;
        background: #FFFFFF;
        border: 1px solid #E0E0E0;
        box-sizing: border-box;
        border-radius: 6px;
        margin-left: 0px;
      }
    } 
  }
  &_data{
    width: 679px;
    height: 319px;  
    display: flex; 
    flex-direction: column;
    align-items: flex-start;
    margin-top: 40px;
    >p:first-child{
      font-family: Roboto;
      font-style: normal;
      font-weight: normal;
      font-size: 36px;
      line-height: 147.3%;
      color: #1B1B1B;
      margin-left: 0px;
      margin-bottom: 30px;
    }
    >P{
      font-family: Roboto;
      font-style: normal;
      font-weight: normal;
      font-size: 14px;
      line-height: 16px;
      color: #1B1B1B;
      margin-top: 0;
      margin-bottom: 20px;
    }
    &_number{
      width: 155px;
      height: 44px;background: #FFFFFF;
      border: 1px solid #E0E0E0;
      box-sizing: border-box;
      border-radius: 6px;
      margin-left: 163px; 
      padding-left: 12px;
    }
    &_date{
      width: 155px;
      height: 44px;
      background: #FFFFFF;
      border: 1px solid #E0E0E0;
      box-sizing: border-box;
      border-radius: 6px;
      margin-left: 173px;
      padding-left: 12px; 
      background:url(../img/calendar.svg) no-repeat center right 15px;
    }
    &_about{
      width: 416px;
      height: 44px;
      background: #FFFFFF;
      border: 1px solid #E0E0E0;
      box-sizing: border-box;
      border-radius: 6px;
      margin-left: 186px; 
      padding-left: 12px;
    }
    &_cod{
      width: 155px;
      height: 44px;
      background: #FFFFFF;
      border: 1px solid #E0E0E0;
      box-sizing: border-box;
      border-radius: 6px;
      margin-left: 129px;
      padding-left: 12px; 
    }
    @include mobile {
      width: 335px;
      >p:first-child{
        font-family: Roboto;
        font-style: normal;
        font-weight: 500;
        font-size: 24px;
        line-height: 120%;
        color: #1B1B1B;
      }
      >p{
        font-family: Roboto;
        font-style: normal;
        font-weight: normal;
        font-size: 14px;
        line-height: 16px;
        color: #1B1B1B;
      }
      &_number{
        width: 335px;
        height: 44px;background: #FFFFFF;
        border: 1px solid #E0E0E0;
        box-sizing: border-box;
        border-radius: 6px;
        margin-left: 0px; 
      }
      &_date{
        width: 335px;
        height: 44px;
        background: #FFFFFF;
        border: 1px solid #E0E0E0;
        box-sizing: border-box;
        border-radius: 6px;
        margin-left: 0px; 
      }
      &_about{
        width: 335px;
        height: 44px;
        background: #FFFFFF;
        border: 1px solid #E0E0E0;
        box-sizing: border-box;
        border-radius: 6px;
        margin-left: 0px; 
      }
      &_cod{
        width: 335px;
        height: 44px;
        background: #FFFFFF;
        border: 1px solid #E0E0E0;
        box-sizing: border-box;
        border-radius: 6px;
        margin-left: 0px; 
      }
    }
  }
  &_info{
    width: 500px;
    height: 191px;  
    margin-top: 40px;
    margin-bottom: 60px;
    >p:first-child{
      font-family: Roboto;
      font-style: normal;
      font-weight: normal;
      font-size: 36px;
      line-height: 147.3%;
      color: #1B1B1B;
      margin-left: 0px;
      margin-bottom: 30px;
    }
    >P{
      font-family: Roboto;
      font-style: normal;
      font-weight: normal;
      font-size: 14px;
      line-height: 16px;
      color: #1B1B1B;
      margin-top: 0;
      margin-bottom: 20px;
    }
    &_number{
      width: 155px;
      height: 44px;
      background: #FFFFFF;
      border: 1px solid #E0E0E0;
      box-sizing: border-box;
      border-radius: 6px;
      margin-left:163px;
      padding-left: 12px;
      }
    &_date{
      width: 155px;
      height: 44px;
      background: #FFFFFF;
      border: 1px solid #E0E0E0;
      box-sizing: border-box;
      border-radius: 6px;
      margin-left:173px;
      padding-left: 12px;
      background:url(../img/calendar.svg) no-repeat center right 15px;

    }
    @include mobile{
      width: 335px;
      >p:first-child{
        font-family: Roboto;
        font-style: normal;
        font-weight: 500;
        font-size: 24px;
        line-height: 120%;
        color: #1B1B1B;
      }
      >p{
        font-family: Roboto;
        font-style: normal;
        font-weight: normal;
        font-size: 14px;
        line-height: 16px;
        color: #1B1B1B;
      }
      &_number{
        width: 335px;
        height: 44px;
        background: #FFFFFF;
        border: 1px solid #E0E0E0;
        box-sizing: border-box;
        border-radius: 6px;
        margin-left:0px;
        }
        &_date{
          width: 335px;
          height: 44px;
          background: #FFFFFF;
          border: 1px solid #E0E0E0;
          box-sizing: border-box;
          border-radius: 6px;
          margin-left:0px;    
        }
    }
  }
  @include mobile {
    align-items: center;
    justify-content: center;
  }
  &_password{
    width: 679px;
    height: 191px;    
    margin-bottom: 174px;
   >p:first-child{
      font-family: Roboto;
      font-style: normal;
      font-weight: normal;
      font-size: 36px;
      line-height: 147.3%;
      color: #1B1B1B;
      margin-left: 0px;
      margin-bottom: 30px;
    }
    >P{
      font-family: Roboto;
      font-style: normal;
      font-weight: normal;
      font-size: 14px;
      line-height: 16px;
      color: #1B1B1B;
      margin-top: 0;
      margin-bottom: 20px;
    }
    &_curr{
        width: 416px;
        height: 44px;
        background: #FFFFFF;
        border: 1px solid #E0E0E0;
        box-sizing: border-box;
        border-radius: 6px;
        margin-left: 128px;
        padding-left: 12px;
        background:url(../img/password.svg) no-repeat center right 15px;
        &.active{        
          border: 2px solid #EB5757;
          outline-color: #EB5757;
        }
    }
    &_conf{
      width: 416px;
      height: 44px;
      background: #FFFFFF;
      border: 1px solid #E0E0E0;
      box-sizing: border-box;
      border-radius: 6px;
      margin-left: 139px;
      padding-left: 12px;
      background:url(../img/password.svg) no-repeat center right 15px;
      &.active{        
        border: 2px solid #EB5757;
        outline-color: #EB5757;
      }
    }
    &_warning{
      display: block;
      font-family: Roboto;
      font-style: normal;
      font-weight: normal;
      font-size: 14px;
      line-height: 16px;
      color: #EB5757;
      margin-left: 260px;
      margin-bottom: 20px;
    }    
  }
}
.continue{
  height: 84px;
  width: 100vw;
  position: fixed;
  bottom: 0;
  background: #FFFFFF;
  box-shadow: 0px -1px 0px #DEE2E4;
    &_button{
    margin:17px auto;
    display: flex;
    flex-direction: row;
    align-items: center;
    padding: 16px 40px;
    width: 176px;
    height: 51px;
    background: #E6E6E6;
    border-radius: 6px;
    font-family: Roboto;
    font-style: normal;
    font-weight: 500;
    font-size: 16px;
    line-height: 19px;
    color: #B4B5BD;
    border-style: none; 
  }
  @include mobile{
    width: 375px;
    height: 84px;
  }
  &_button{
    margin:17px auto;
    display: block;
    align-items: center;
    padding: 16px 40px;
    width: 335px;
    height: 51px;
    background: #61A199;
    border-radius: 6px;
    font-family: Roboto;
    font-style: normal;
    font-weight: 500;
    font-size: 16px;
    line-height: 19px;
    color:#FFFFFF;
    border-style: none; 
    text-align: center;
    
  
  }
  &_button:disabled{
    background: #E6E6E6;
    color: #B4B5BD;
  }   
}  