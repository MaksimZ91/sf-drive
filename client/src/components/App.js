import  React from "react";
import Header from "./header/header"
import Aboutpage from "./pages/about/aboutPage"
import Mainpage from "./pages/main/mainPage"
import Faq from "./pages/faq/Faq"
import Registpage from "./pages/registr/registPage"
import {Route, BrowserRouter, Switch} from "react-router-dom"
import {FormProvider} from './contextApp'

function App () {
  
  return (  
    <FormProvider>
    <BrowserRouter>
    <Header/>
    <Switch>
    <Route path="/" component={Mainpage} exact/>
    <Route path="/about" component={Aboutpage} exact/>
    <Route path="/registr" component={Registpage}/>
    <Route path="/faq" component={Faq}/>  
    </Switch> 
    </BrowserRouter>  
    </FormProvider>   
    
  ) 
          
    
}

export default App;