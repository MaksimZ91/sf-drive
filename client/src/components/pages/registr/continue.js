import React, { useContext} from "react"
import { RegistContext } from './formContex'
import {useHttp} from '../../../hooks/http.hook'
import {useAuth} from '../../../hooks/autn.hook'

function Continue (props){
  const {form, formValidation} = useContext(RegistContext)
  const {login}=useAuth()
  const {request} = useHttp()
 console.log(form)

  const authorRequest = async () => {
    try {
        const data = await request('http://localhost:5000/registr','POST',{...form})     
        props.value.message(false)        
    } catch (e) {
      console.log(e)
      props.value.message(true)                 
    }    
}
   return (
    <>
    <section className={(props.value.recovery||props.value.openAuthor)?'continue active':"continue"} >
    <input className="continue_button" onClick={authorRequest} type="submit" name="submit" value="Продолжить" disabled={formValidation}  />
    </section> 
    </>
  )
}
export default Continue