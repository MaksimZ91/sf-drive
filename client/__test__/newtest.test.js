import React from 'react'
import { render, screen } from '@testing-library/react'
import { shallow, mount } from 'enzyme'
import { FormProvider } from '../src/components/contextApp'
import { BrowserRouter } from 'react-router-dom'
import App from '../src/components/App'
import Header from '../src/components/header/header'
import Navbar from '../src/components/header/navbar'
import Password from '../src/components/pages/registr/formPassword'
import { FormRefisrtProvider } from '../src/components/pages/registr/formContex'




const fn = () => {}

describe('App test', () => {

  it('App render test', () => {
    render(<FormProvider><App/></FormProvider>)
    const LinkElement = screen.getByText(/Частые вопросы/i)
    expect(LinkElement).toBeInTheDocument()
  }) 

  describe('Header tests', ()=>{
    it('Header to be defined', () => {
      const component = mount(<FormProvider><BrowserRouter><Header/></BrowserRouter></FormProvider>)
      expect(component).toBeDefined()
      })
  
    it('Click button login', () => {
      const  mockClick = jest.fn()
      const props = {
       value : {
         diss:false,
         rec:false,
         openAuth:jest.fn()
         
       }
      }
      const element = mount(<FormProvider><BrowserRouter><Navbar  {...props} /></BrowserRouter></FormProvider>)  
      element.find('.header__button').instance().handelClick = mockClick
      element.find('.header__button').instance().handelClick()
      element.find('.header__button').simulate('click')
      expect(mockClick).toHaveBeenCalled()
      expect(props.value.openAuth).toHaveBeenCalled()
    })
  })

  describe('Registr form testing', () => {
    it('Changer value form testing', () => {    
      const wrapper = mount(<FormRefisrtProvider><Password/></FormRefisrtProvider>) 
      wrapper.setState({ pass: 'bar' })
      wrapper.find('.registr_about_password_curr').simulate('change', event)   
      expect(state.pass).toEqual('password')

    })
  })

})



