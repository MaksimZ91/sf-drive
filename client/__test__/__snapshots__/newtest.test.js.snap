// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test App error test 1`] = `
<Fragment>
  <div
    className="error"
  >
    <p>
      error
    </p>
  </div>
</Fragment>
`;

exports[`App test App success test 1`] = `
<FormProvider>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Provider
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
      >
        <NotBookingPage>
          <Sucses
            text="Test"
          >
            <div
              className="sucses"
            >
              <p>
                Test
              </p>
            </div>
          </Sucses>
          <main
            className="booking"
          >
            <div
              className="booking_wrapper"
            >
              <img
                alt="road"
                className="booking_wrapper_img"
                src="./src/img/road.svg"
              />
              <p
                className="booking_wrapper_text"
              >
                Здесь будут ваши бронирования
              </p>
            </div>
          </main>
        </NotBookingPage>
      </Provider>
    </Router>
  </BrowserRouter>
</FormProvider>
`;

exports[`App test Form testing Registr changer value form testing 1`] = `
<FormRegisrtProvider>
  <Password>
    <form
      className="registr_about_password"
    >
      <p>
        Пароль
      </p>
      <p>
        Придумайте пароль
        <input
          className="registr_about_password_curr"
          name="pass"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          placeholder="•••••••••••••••••••"
          type="text"
          value="Hello"
        />
      </p>
      <p>
        Повторите пароль
        <input
          className="registr_about_password_conf"
          name="secondPassword"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          placeholder="•••••••••••••••••••"
          type="text"
          value=""
        />
      </p>
    </form>
  </Password>
</FormRegisrtProvider>
`;

exports[`App test Header tests Click button login 1`] = `
<FormProvider>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Navbar
        value={
          Object {
            "diss": false,
            "openAuth": [MockFunction] {
              "calls": Array [
                Array [
                  true,
                ],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            },
            "rec": false,
          }
        }
      >
        <div
          className="nav_wrapper"
        >
          <nav
            className="header__nav"
          >
            <ul
              className="header__list"
            >
              <button
                onClick={[Function]}
              >
                Тест
              </button>
              <li
                className="header__link"
              >
                <NavLink
                  to="/about"
                >
                  <Link
                    aria-current={null}
                    to={
                      Object {
                        "hash": "",
                        "pathname": "/about",
                        "search": "",
                        "state": null,
                      }
                    }
                  >
                    <LinkAnchor
                      aria-current={null}
                      href="/about"
                      navigate={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/about"
                        onClick={[Function]}
                      >
                        О нас
                      </a>
                    </LinkAnchor>
                  </Link>
                </NavLink>
              </li>
              <li
                className="header__link"
              >
                <NavLink
                  to="/recovery"
                >
                  <Link
                    aria-current={null}
                    to={
                      Object {
                        "hash": "",
                        "pathname": "/recovery",
                        "search": "",
                        "state": null,
                      }
                    }
                  >
                    <LinkAnchor
                      aria-current={null}
                      href="/recovery"
                      navigate={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/recovery"
                        onClick={[Function]}
                      >
                        Условия
                      </a>
                    </LinkAnchor>
                  </Link>
                </NavLink>
              </li>
              <li
                className="header__link"
              >
                <NavLink
                  to="/faq"
                >
                  <Link
                    aria-current={null}
                    to={
                      Object {
                        "hash": "",
                        "pathname": "/faq",
                        "search": "",
                        "state": null,
                      }
                    }
                  >
                    <LinkAnchor
                      aria-current={null}
                      href="/faq"
                      navigate={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/faq"
                        onClick={[Function]}
                      >
                        Частые вопросы
                      </a>
                    </LinkAnchor>
                  </Link>
                </NavLink>
              </li>
            </ul>
          </nav>
          <button
            className="header__button"
            disabled={false}
            onClick={[Function]}
          >
            Войти
          </button>
        </div>
      </Navbar>
    </Router>
  </BrowserRouter>
</FormProvider>
`;

exports[`App test Header tests Header to be defined 1`] = `
<FormProvider>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Header>
        <header
          className="header"
        >
          <Logo>
            <div
              className="header__logo"
            >
              <a
                aria-label="Вы сфокусированы на ссылки которая ведет на страницу SkillDrive"
                href="/"
              >
                <img
                  alt="logo SkillDrive"
                  src="../src/img/Logo.svg"
                />
              </a>
            </div>
          </Logo>
          <Navbar
            value={
              Object {
                "diss": false,
                "openAuth": [Function],
                "openMobileMenu": false,
                "rec": false,
              }
            }
          >
            <div
              className="nav_wrapper"
            >
              <nav
                className="header__nav"
              >
                <ul
                  className="header__list"
                >
                  <button
                    onClick={[Function]}
                  >
                    Тест
                  </button>
                  <li
                    className="header__link"
                  >
                    <NavLink
                      to="/about"
                    >
                      <Link
                        aria-current={null}
                        to={
                          Object {
                            "hash": "",
                            "pathname": "/about",
                            "search": "",
                            "state": null,
                          }
                        }
                      >
                        <LinkAnchor
                          aria-current={null}
                          href="/about"
                          navigate={[Function]}
                        >
                          <a
                            aria-current={null}
                            href="/about"
                            onClick={[Function]}
                          >
                            О нас
                          </a>
                        </LinkAnchor>
                      </Link>
                    </NavLink>
                  </li>
                  <li
                    className="header__link"
                  >
                    <NavLink
                      to="/recovery"
                    >
                      <Link
                        aria-current={null}
                        to={
                          Object {
                            "hash": "",
                            "pathname": "/recovery",
                            "search": "",
                            "state": null,
                          }
                        }
                      >
                        <LinkAnchor
                          aria-current={null}
                          href="/recovery"
                          navigate={[Function]}
                        >
                          <a
                            aria-current={null}
                            href="/recovery"
                            onClick={[Function]}
                          >
                            Условия
                          </a>
                        </LinkAnchor>
                      </Link>
                    </NavLink>
                  </li>
                  <li
                    className="header__link"
                  >
                    <NavLink
                      to="/faq"
                    >
                      <Link
                        aria-current={null}
                        to={
                          Object {
                            "hash": "",
                            "pathname": "/faq",
                            "search": "",
                            "state": null,
                          }
                        }
                      >
                        <LinkAnchor
                          aria-current={null}
                          href="/faq"
                          navigate={[Function]}
                        >
                          <a
                            aria-current={null}
                            href="/faq"
                            onClick={[Function]}
                          >
                            Частые вопросы
                          </a>
                        </LinkAnchor>
                      </Link>
                    </NavLink>
                  </li>
                </ul>
              </nav>
              <button
                className="header__button"
                disabled={false}
                onClick={[Function]}
              >
                Войти
              </button>
            </div>
          </Navbar>
          <img
            alt="mobile_menu"
            className="header_mobile_menu"
            onClick={[Function]}
            src="../src/img/mobile_menu.svg"
          />
        </header>
      </Header>
    </Router>
  </BrowserRouter>
</FormProvider>
`;
